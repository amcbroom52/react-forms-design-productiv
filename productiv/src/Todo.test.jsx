import { describe, it, expect } from "vitest";

import React from "react";
import { render } from "@testing-library/react";
import Todo from "./Todo";

const TODO = {
    id: '1',
    title: "title",
    description: "description",
    priority: 1
};

describe("Todo component", function () {
    it("renders without crashing", function () {
        render(<Todo todo={TODO} />);
    });

    it("matches snapshot", function () {
        const { asFragment } = render(<Todo todo={TODO} />);
        expect(asFragment()).toMatchSnapshot();
    })

    it("contains Todo values", function() {
        const { container } = render(<Todo todo={TODO} />);
        expect(container).toContainHTML("description");
        expect(container).toContainHTML("title");
        expect(container).toContainHTML("Priority: 1");
    })
});
