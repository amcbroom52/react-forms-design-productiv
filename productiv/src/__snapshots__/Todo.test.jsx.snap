// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Todo component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
    id="1"
  >
    <div>
      <b>
        title
      </b>
       
      <small>
        Priority: 1
      </small>
    </div>
    <div>
      <small>
        description
      </small>
    </div>
  </div>
</DocumentFragment>
`;
